OSDOCS#5174: Azure: Document installing OpenShift in a restricted network [Installer-provisioned] (#61086)
openshift/openshift-docs: modules/installation-azure-config-yaml.adoc

apiVersion: v1
baseDomain: example.com <1>
controlPlane: <2>
ifndef::azure-restricted[]
  hyperthreading: Enabled <3> <4>
endif::azure-restricted[]
ifdef::azure-restricted[]
  hyperthreading: Enabled <3>
endif::azure-restricted[]
  name: master
  platform:
    azure:
      encryptionAtHost: true
      ultraSSDCapability: Enabled
      osDisk:
ifndef::azure-restricted[]
        diskSizeGB: 1024 <5>
endif::azure-restricted[]
ifdef::azure-restricted[]
        diskSizeGB: 1024 <4>
endif::azure-restricted[]
        diskType: Premium_LRS
        diskEncryptionSet:
          resourceGroup: disk_encryption_set_resource_group
          name: disk_encryption_set_name
          subscriptionId: secondary_subscription_id
      type: Standard_D8s_v3
  replicas: 3
ifndef::azure-restricted[]
compute: <2>
endif::azure-restricted[]
ifdef::azure-restricted[]
compute: <5>

> endif::azure-restricted[]

Reviewer: The description of <2> can be used here. No need to add ifdef and ifndef for `azure-restricted`.

    region: usgovvirginia
endif::gov[]
    resourceGroupName: existing_resource_group <9>

>     networkResourceGroupName: vnet_resource_group <10>

Reviewer 2: I checked that <10> ~ <13> are repeated twice in private cluster installation doc https://61086--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installing-azure-private.html#installation-azure-config-yaml_installing-azure-private

Reviewer: Lines 107-110 needs to be removed. They are getting repeated in lines 112-115.

    virtualNetwork: vnet <11>
    controlPlaneSubnet: control_plane_subnet <12>

>     computeSubnet: compute_subnet <13>

Reviewer: There are no ifdef and endif for these lines related to `azure-restricted`. These lines will be included in all the other azure documentations.

<8> Specify the name of the resource group that contains the DNS zone for your base domain.
<9> Specify the name of an already existing resource group to install your cluster to. If undefined, a new resource group is created for the cluster.
ifdef::vnet,private,gov,azure-restricted[]
<10> If you use an existing VNet, specify the name of the resource group that contains it.
<11> If you use an existing VNet, specify its name.
<12> If you use an existing VNet, specify the name of the subnet to host the control plane machines.
<13> If you use an existing VNet, specify the name of the subnet to host the compute machines.
endif::vnet,private,gov,azure-restricted[]
ifdef::private,gov[]
<14> You can customize your own outbound routing. Configuring user-defined routing prevents exposing external endpoints in your cluster. User-defined routing for egress requires deploying your cluster to an existing VNet.
endif::private,gov[]
ifdef::gov[]
<15> Specify the name of the Azure cloud environment to deploy your cluster to. Set `AzureUSGovernmentCloud` to deploy to a Microsoft Azure Government (MAG) region. The default value is `AzurePublicCloud`.
endif::gov[]
ifdef::azure-restricted[]

> <14> When using Azure Firewall to restrict Internet access, you must configure outbound routing to send traffic through the Firewall. Configuring user-defined routing prevents exposing external endpoints in your cluster. User-defined routing for egress requires deploying your cluster to an existing VNet.

Author: Last sentence can be excluded. Ack'ed

====
<16> You can optionally provide the `sshKey` value that you use to access the machines in your cluster.
endif::openshift-origin,azure-restricted[]
ifdef::openshift-origin[]
<15> You can optionally provide the `sshKey` value that you use to access the machines in your cluster.
endif::openshift-origin[]
endif::private[]
ifdef::gov[]
ifndef::openshift-origin,azure-restricted[]
<16> Whether to enable or disable FIPS mode. By default, FIPS mode is not enabled. If FIPS mode is enabled, the {op-system-first} machines that {product-title} runs on bypass the default Kubernetes cryptography suite and use the cryptography modules that are provided with {op-system} instead.
+
[IMPORTANT]
====
The use of FIPS Validated / Modules in Process cryptographic libraries is only supported on {product-title} deployments on the `x86_64` architecture.
====
<17> You can optionally provide the `sshKey` value that you use to access the machines in your cluster.
endif::openshift-origin,azure-restricted[]
ifdef::openshift-origin[]
<16> You can optionally provide the `sshKey` value that you use to access the machines in your cluster.
endif::openshift-origin[]
endif::gov[]
ifndef::vnet,private,gov[]

> ifndef::openshift-origin,azure-restricted[]

Author: Yes. Addressed

endif::azure-restricted[]
//To configure faster storage for etcd, especially for larger clusters, set the
//storage type as `io1` and set `iops` to `2000`.
<6> Specify a list of zones to deploy your machines to. For high availability, specify at least two zones.
<7> The cluster network plugin to install. The supported values are `OVNKubernetes` and `OpenShiftSDN`. The default value is `OVNKubernetes`.
<8> Specify the name of the resource group that contains the DNS zone for your base domain.
<9> Specify the name of an already existing resource group to install your cluster to. If undefined, a new resource group is created for the cluster.
ifdef::vnet,private,gov,azure-restricted[]
<10> If you use an existing VNet, specify the name of the resource group that contains it.
<11> If you use an existing VNet, specify its name.
<12> If you use an existing VNet, specify the name of the subnet to host the control plane machines.
<13> If you use an existing VNet, specify the name of the subnet to host the compute machines.
endif::vnet,private,gov,azure-restricted[]
ifdef::private,gov,azure-restricted[]

> <14> You can customize your own outbound routing. Configuring user-defined routing prevents exposing external endpoints in your cluster. User-defined routing for egress requires deploying your cluster to an existing VNet.

Reviewer: The content for restricted network will be different:

`When using Azure Firewall to restrict Internet access, outbound routing is needed to be configured to be sent via the Firewall. Configuring user-defined routing prevents exposing external endpoints in your cluster. User-defined routing for egress requires deploying your cluster to an existing VNet.`

endif::gov[]
----
ifndef::gov[]
<1> Required. The `baseDomain` parameter specifies the base domain of your cloud provider. The installation program prompts you for this value.
endif::gov[]
ifdef::gov[]
<1> Required.
endif::gov[]
ifndef::azure-restricted[]
<2> If you do not provide these parameters and values, the installation program provides the default value.
endif::azure-restricted[]
ifdef::azure-restricted[]
<2> The configuration for the machines that comprise the control plane. The `controlPlane` section is a single mapping. The first line of the `controlPlane` section must not begin with a hyphen, `-`. You can use only one control plane pool. If you do not provide these parameters and values, the installation program provides the default value.
endif::azure-restricted[]

> ifndef::azure-restricted[]

Reviewer: No need to add these lines. We can reuse the description of  <2>.

Reviewer 2: +1

+
[IMPORTANT]
====
If you disable simultaneous multithreading, ensure that your capacity planning accounts for the dramatically decreased machine performance. Use larger virtual machine types, such as `Standard_D8s_v3`, for your machines if you disable simultaneous multithreading.
====
endif::azure-restricted[]
ifndef::azure-restricted[]
<4> Whether to enable or disable simultaneous multithreading, or `hyperthreading`. By default, simultaneous multithreading is enabled to increase the performance of your machines' cores. You can disable it by setting the parameter value to `Disabled`. If you disable simultaneous multithreading in some cluster machines, you must disable it in all cluster machines.
+
[IMPORTANT]
====
If you disable simultaneous multithreading, ensure that your capacity planning accounts for the dramatically decreased machine performance. Use larger virtual machine types, such as `Standard_D8s_v3`, for your machines if you disable simultaneous multithreading.
====
endif::azure-restricted[]

> ifdef::azure-restricted[]

Reviewer: I checked other Azure installation documentations and the 4th point is included in them. You can include that in this documentation as well.

<8> Specify the name of the resource group that contains the DNS zone for your base domain.
<9> Specify the name of an already existing resource group to install your cluster to. If undefined, a new resource group is created for the cluster.
ifdef::vnet,private,gov,azure-restricted[]
<10> If you use an existing VNet, specify the name of the resource group that contains it.
<11> If you use an existing VNet, specify its name.
<12> If you use an existing VNet, specify the name of the subnet to host the control plane machines.
<13> If you use an existing VNet, specify the name of the subnet to host the compute machines.
endif::vnet,private,gov,azure-restricted[]
ifdef::private,gov[]
<14> You can customize your own outbound routing. Configuring user-defined routing prevents exposing external endpoints in your cluster. User-defined routing for egress requires deploying your cluster to an existing VNet.
endif::private,gov[]
ifdef::gov[]
<15> Specify the name of the Azure cloud environment to deploy your cluster to. Set `AzureUSGovernmentCloud` to deploy to a Microsoft Azure Government (MAG) region. The default value is `AzurePublicCloud`.
endif::gov[]
ifdef::azure-restricted[]

> <14> When using Azure Firewall to restrict Internet access, you must configure outbound routing to be sent by using the Firewall. Configuring user-defined routing prevents exposing external endpoints in your cluster. User-defined routing for egress requires deploying your cluster to an existing VNet.

Reviewer: ```suggestion
<14> When using Azure Firewall to restrict Internet access, you must configure outbound routing to send traffic through the Firewall. Configuring user-defined routing prevents exposing external endpoints in your cluster. User-defined routing for egress requires deploying your cluster to an existing VNet.
```

endif::openshift-origin,azure-restricted[]
ifdef::openshift-origin[]
<16> You can optionally provide the `sshKey` value that you use to access the machines in your cluster.
endif::openshift-origin[]
endif::gov[]
ifndef::vnet,private,gov[]
ifndef::openshift-origin,azure-restricted[]
<10> Whether to enable or disable FIPS mode. By default, FIPS mode is not enabled. If FIPS mode is enabled, the {op-system-first} machines that {product-title} runs on bypass the default Kubernetes cryptography suite and use the cryptography modules that are provided with {op-system} instead.
+
[IMPORTANT]
====
The use of FIPS Validated / Modules in Process cryptographic libraries is only supported on {product-title} deployments on the `x86_64` architecture.
====
<11> You can optionally provide the `sshKey` value that you use to access the machines in your cluster.
endif::openshift-origin,azure-restricted[]
ifndef::azure-restricted[]
ifdef::openshift-origin[]
<10> You can optionally provide the `sshKey` value that you use to access the machines in your cluster.
endif::openshift-origin[]
endif::azure-restricted[]
endif::vnet,private,gov[]
+

> [NOTE]

Reviewer: This Note should not be displayed again for Azure restricted network. It is already displayed above.

Reviewer: This Note is still getting displayed twice. To rectify the issue you can move the following lines after this Note and remove lines 244 to 248:
```
<17> Provide the contents of the certificate file that you used for your mirror registry.
<18> Provide the `imageContentSources` section from the output of the command to mirror the repository.
<19> How to publish the user-facing endpoints of your cluster. When using Azure Firewall to restrict Internet access, set `publish` to `Internal` to deploy a private cluster. The user-facing endpoints then cannot be accessed from the internet. The default value is `External`.
```
